staging/src/k8s.io/cli-runtime/pkg/resource/dry_run_verifier.go

```diff
@@ -29,16 +29,23 @@ import (
 
 // VerifyDryRun returns nil if a resource group-version-kind supports
 // server-side dry-run. Otherwise, an error is returned.
-func VerifyDryRun(gvk schema.GroupVersionKind, dynamicClient dynamic.Interface, discoveryClient discovery.DiscoveryInterface) error {
-	verifier := NewDryRunVerifier(dynamicClient, discoveryClient)
+func VerifyDryRun(gvk schema.GroupVersionKind, dynamicClient dynamic.Interface, openAPIDoc *openapi_v2.Document) error {
+	verifier := &DryRunVerifier{
+		finder:     NewCRDFinder(CRDFromDynamic(dynamicClient)),
+		openAPIDoc: openAPIDoc,
+	}
 	return verifier.HasSupport(gvk)
 }
 
-func NewDryRunVerifier(dynamicClient dynamic.Interface, discoveryClient discovery.DiscoveryInterface) *DryRunVerifier {
-	return &DryRunVerifier{
-		finder:        NewCRDFinder(CRDFromDynamic(dynamicClient)),
-		openAPIGetter: discoveryClient,
+func NewDryRunVerifier(dynamicClient dynamic.Interface, openAPIGetter discovery.OpenAPISchemaInterface) (*DryRunVerifier, error) {
```

Comment:
Why is this not taking the `openapi_v2.Document` directly? I'm wondering if this is going to cause some double-download of the OpenAPI?

Reply:
we should modify all of kubectl cmd that calls this function

Reply:
Are there so many calls?

Reply:
Yeah there are a lot of calls. I'll update one, but I want to make sure this looks okay before we update all of them

Reply:
I think if you want to cache the document here, you can still use discoveryClient discovery.DiscoveryInterface et continue to cache an openapi, to pass it to DryRunVerifier struct

Reply:
but I think you have to choose between caching the document in the Verifier 'constructor' ans continue using DiscoveryInterface or modify the constructor of the verifier to accept an OpenAPISchemaInterface the cache the call to the api. As I see you are using both in the same time

Reply:
PTAL, each subcommand is updated and this change builds properly